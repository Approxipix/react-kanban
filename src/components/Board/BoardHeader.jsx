import React, { Component } from 'react';
import { connect } from "react-redux";
import { bindActionCreators } from 'redux';
import { deleteBoard } from "../../redux/boardReducer/actions";
import { history } from "../../redux/store";
import styled from "styled-components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import BoardTitleEdit from "./BoardTitleEdit";
import ColorPicker from "../ColorPicker";

const Header = styled.div`
  margin-bottom: 1rem;
  padding: 0 1.5rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

const Actions = styled.div`
  display: flex;
  align-items: center;
  &:not(:last-child) {
    border-right: 1px solid red;
  }
`;

const Button = styled.button`
  margin: 0.5rem;
  padding: .5rem;
  cursor: pointer;
  color: #fff;
  border-radius: .2rem;
  transition: background .2s ease-in-out;
  white-space: nowrap;
  &:hover {
    background: rgba(0, 0, 0, 0.2);
  }
`;

const Title = styled.h1`
  margin: 0;
  padding: .3rem .5rem;
  font-size: 1.3rem;
  color: #fff;
  font-weight: 600;
`;

const VerticalLine = styled.div`
  width: 1px;
  height: 1.3rem;
  background-color: #d6d6d6;
`;

class BoardHeader extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isEditing: false,
    }
  }

  deleteBoard = () => {
    const { board, actions } = this.props;
    actions.deleteBoard({ _boardId: board._boardId });
    history.push('/boards')
  };

  toggleEditing = () => {
    this.setState({
      isEditing: !this.state.isEditing,
    })
  };

  render() {
    const { isEditing } = this.state;
    const { board } = this.props;
    return (
      <Header>
        {!isEditing ? (
          <Title onClick={() => this.toggleEditing()}>
            {board.title}
          </Title>
        ) : (
          <BoardTitleEdit toggleEditing={this.toggleEditing}/>
        )}
        <Actions>
          <ColorPicker />
          <VerticalLine />
          <Button onClick={() => this.deleteBoard()}>
            <FontAwesomeIcon icon="trash" />
            &nbsp;Delete board
          </Button>
        </Actions>
      </Header>
    )
  }
}

function mapStateToProps(state) {
  const { rootReducer } = state;
  return {
    board: rootReducer.boards[rootReducer.currentBoardIndex],
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      deleteBoard: deleteBoard,
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(BoardHeader);

