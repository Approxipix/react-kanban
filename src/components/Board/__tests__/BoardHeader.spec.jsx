import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import BoardHeader from '../BoardHeader';

const mockStore = configureStore([]);

describe('<BoardHeader>', () => {
  let store;
  let component;
  let mockToggleSidebar = jest.fn();

  beforeEach(() => {
    store = mockStore({});
    store.dispatch = mockToggleSidebar;

    const wrapper = shallow(<BoardHeader store={store}  />);
    component = wrapper.find('BoardHeader').dive();
  });

  it('should render correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should dispatch an action after click on button", () => {
    const expectedAction = {
      type: 'TOGGLE_SIDEBAR',
    };

    component.find('Button').simulate('click');

    expect(mockToggleSidebar).toHaveBeenCalledTimes(1);
    expect(mockToggleSidebar).toHaveBeenCalledWith(expectedAction);
  });
});

