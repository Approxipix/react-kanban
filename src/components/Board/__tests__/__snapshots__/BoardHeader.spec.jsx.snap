// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoardHeader> should render correctly 1`] = `
<styled.div>
  <Connect(BoardTitle) />
  <Button
    onClick={[Function]}
  >
    Show menu
  </Button>
</styled.div>
`;
